### PR TITLE
[netcore] Implement some of the managed synchronization classes in managed code using the corresponding code from CoreRT.

### PR DESCRIPTION
**Pull request by vargaz**
The CoreRT classes are from rev b2270a3303633312236be3a6ec78d68f856e9b76.

### CODE REVIEW

File: netcore/CoreFX.issues.rsp [hunk 1]
```diff
@@ -937,3 +937,8 @@
 # Requires NativeLibrary implementation
 -nomethod System.Net.Tests.HttpWebRequestTest.ServicePoint_GetValue_ExpectedResult
 
+# Not supported on mono
```
Comment (jaykrell) [inline netcore/CoreFX.issues.rsp]: Not even on Windows?

**Comment by jaykrell** [comment]
I realize this allows replacing a bunch of native code on Unix, but does it make sense on Windows too?
Isn't there a choice on Windows of pretty thin wrapping of OS?

**Comment by vargaz** [comment]
The failures are relevant.

**Comment by filipnavara** [comment]
@jaykrell That's already the case. On Windows it's mostly P/Invokes.

**Comment by vargaz** [comment]
Closing for now, maybe somebody can resurrect this later.

**Pull request closed without merging**